tests/test_core.py

```diff
@@ -1171,43 +1171,74 @@ def test_yin_tone(freq):
+    f0 = f0[:-2]
+
     target_f0 = np.load(os.path.join("tests", "data", "pitch-yin.npy"))
     assert np.allclose(np.log2(f0), np.log2(target_f0), rtol=0, atol=1e-2)
 
 
+def test_yin_chirp_instant():
+    # test yin on a chirp, using frame-wise instantaneous frequency as ground truth
+    sr = 22050
+    chirp_min, chirp_max = 220, 640
+
+    t = np.arange(sr) / sr
+    f = chirp_min * (chirp_max / chirp_min) ** t
+
+    fl = 2048
+    hl = 512
+
+    y = librosa.chirp(fmin=chirp_min, fmax=chirp_max, sr=sr, duration=1.0, linear=False)
+    target_f0 = librosa.util.frame(f, frame_length=fl, hop_length=hl).mean(axis=0)
+
+    f0 = librosa.yin(y, fmin=110, fmax=880, sr=sr, frame_length=fl, hop_length=hl, center=False)
+    assert np.allclose(np.log2(f0), np.log2(target_f0), rtol=0, atol=1e-2)
+
+
 @pytest.mark.xfail(raises=librosa.ParameterError)
 @pytest.mark.parametrize(
-    "fmin,fmax,win_length,frame_length",
+    "fmin,fmax,frame_length",
     [
-        (None, None, None, 2048),  # neither
-        (110, None, None, 2048),  # no fmax
-        (None, 880, None, 2048),  # no fmin
-        (110, 880, 2049, 2048),  # win_length >= frame_length
-        (-1, 440, None, 2048),  # Negative fmin
-        (440, 220, None, 2048),  # fmin > fmax
-        (440, 16000, None, 2048),  # fmax > nyquist
-        (10, 21, None, 2048),  # frame_length - win_length - 1 <= sr/fmax
+        (None, None, 2048),  # neither
+        (110, None, 2048),  # no fmax
+        (None, 880, 2048),  # no fmin
+        (-1, 440, 2048),  # Negative fmin
+        (440, 220, 2048),  # fmin > fmax
+        (440, 16000, 2048),  # fmax > nyquist
+        (10, 21, 2048),  # sr / fmin >= frame_length - 1
     ],
 )
-def test_yin_fail(fmin, fmax, win_length, frame_length):
+def test_yin_fail(fmin, fmax, frame_length):
     y = librosa.tone(110, duration=1.0)
     librosa.yin(
-        y, fmin=fmin, fmax=fmax, win_length=win_length, frame_length=frame_length
+        y, fmin=fmin, fmax=fmax, frame_length=frame_length
     )
 
+def test_yin_warn():
+    y = librosa.tone(110, duration=1.0)
+
+    # win_length is deprecated
+    with pytest.warns(FutureWarning, match="deprecated"):
+        librosa.yin(y, fmin=110, fmax=1000, win_length=1024)
+
+    # sr / fmin >= frame_length // 2
+    with pytest.warns(UserWarning, match="two periods"):
+        librosa.yin(y, fmin=20, fmax=1000)
 
 @pytest.mark.parametrize("freq", [110, 220, 440, 880])
 def test_pyin_tone(freq):
     y = librosa.tone(freq, duration=1.0)
     f0, _, _ = librosa.pyin(y, fmin=110, fmax=1000, center=False)
-    # Skip the first frame, since the voicing prior does not allow it
-    assert np.allclose(np.log2(f0[1:]), np.log2(freq), rtol=0, atol=1e-2)
+    assert np.allclose(np.log2(f0), np.log2(freq), rtol=0, atol=1e-2)
 
 
 def test_pyin_multi():
-    y = np.stack([librosa.tone(440, duration=1.0), librosa.tone(560, duration=1.0)])
+    y = np.stack([librosa.tone(440, duration=0.1), librosa.tone(560, duration=0.1)])
```

Comment:
I'm curious why this duration was shortened so much in this test?

Reply:
Oops, sorry. That's a debugging remnant that shouldn't have been committed, I'll change that back.